sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -37,7 +36,8 @@ class SparkOptimizer(
 
   override def earlyScanPushDownRules: Seq[Rule[LogicalPlan]] =
     // TODO: move SchemaPruning into catalyst
-    SchemaPruning :: V2ScanRelationPushDown :: V2Writes :: PruneFileSourcePartitions :: Nil
+    SchemaPruning :: V2ScanRelationPushDown :: V1Writes :: V2Writes ::
+      PruneFileSourcePartitions:: Nil
```

Comment:
It does slightly confuses that `V*Writes` are here. Look, `earlyScanPushDownRules` is about:
" ... projection and filter pushdown to scans"
but V1Writes:
"... makes sure the v1 write requirement, e.g. requiredOrdering" 
something like opposite thing - pulling from instead of pushing down.

Reply:
Here is some history of why `V2Writes ` is at `earlyScanPushDownRules `, https://github.com/apache/spark/pull/30806#discussion_r544466723.

I agree the name is not matched, do you have other better place to go ?